### PR TITLE
Add support for absolute file paths

### PR DESCRIPTION
**Pull request by ca136**
Also adds an example directory where you can test coffeescript changes before compiling.

### CODE REVIEW

File: lib/compile.coffee [hunk 1]
```diff
@@ -199,7 +200,7 @@ compile = (options) ->
         # Strip out any filters (e.g. ng-repeat="foo in bar | limitTo: 10")
         # and split by whitespace and compact the result (remove any empty
         # strings in the list) as well as the 'track by' option in angular
-        repeatExpSplit = _.compact repeatExp
+        repeatExpSplit = repeatExp
```
Comment (ca136) [inline lib/compile.coffee]: @steve8708 I have not fully tested the compiled JS in our project, so you may want to hold off merging until I do that.

Comment (ca136) [inline lib/compile.coffee]: Looks like `_.compact` was unnecessary here.

File: lib/compile.coffee [hunk 2]
```diff
@@ -199,9 +200,11 @@ compile = (options) ->
         # Strip out any filters (e.g. ng-repeat="foo in bar | limitTo: 10")
         # and split by whitespace and compact the result (remove any empty
         # strings in the list) as well as the 'track by' option in angular
-        repeatExpSplit = _.compact repeatExp
+        repeatExp = _.compact repeatExp if typeof repeatExp isnt 'string'
+
+        repeatExpSplit = repeatExp
           .split(' | ')[0]
-          .split('track by')[0]
+          .split('track by')[0].trim()
```
Comment (ca136) [inline lib/compile.coffee]: There was an extra space at the end when you use `track by` in an ng-repeat that was causing compylr to fail.

File: lib/compile.coffee [hunk 3]
```diff
@@ -163,6 +163,7 @@ unescapeTripleBraces = (str) ->
 
 compile = (options) ->
   filePath = argv.file or options.file
+  filePath = filePath?.replace /^\//, '' # replace leading / to read the file
```
Comment (steve8708) [inline lib/compile.coffee]: are you sure we need to do this here? I could see a script passing an absolute path in bc we are reading files - e.g. node often times wants to work in absolute paths so getting a path to a file often results in `/Users/steve/...` in which case we want that preceeding slash

All the other places make sense to me though as those are in ng-includes, but this one is just different in that it is passed directly to fs.readFileSync

Comment (ca136) [inline lib/compile.coffee]: Yeah, I actually don't think we use this so I'll remove it for now.

**Comment by ca136** [comment]
@steve8708 updated and tested the compiled code locally.

**Comment by steve8708** [comment]
nice

**Pull request merged**